src/transformation/visitors/identifier.ts

```diff
@@ -53,9 +54,37 @@ function transformNonValueIdentifier(
         }
     }
 
-    const text = hasUnsafeIdentifierName(context, identifier, symbol)
-        ? createSafeName(identifier.text)
-        : identifier.text;
+    let text = hasUnsafeIdentifierName(context, identifier, symbol) ? createSafeName(identifier.text) : identifier.text;
+
+    const customNames = context.options.customNames as Array<{ name: string; value: string }>;
+
+    if (customNames) {
+        const customName = customNames.find(cN => cN.name === text);
+
+        if (customName) {
+            text = customName.value;
+        }
+    }
```

Comment:
I don't like this at all, besides the fact this is terrible for tstl performance, it is also a paint o maintain and probably causing a bunch of weird unexpected bugs due to things unexpectedly being renamed.

Unless you have a very convincing argument for why we should provide both CLI and annotation options, I would prefer this be left out.

Reply:
Just so I understand you correctly, you are referring to the tsconfig integration? Yeah, I would probably agree with you.

Reply:
Yes, I would remove the CLI and tsConfig part of this MR, and just leave the annotation.

Reply:
> Yes, I would remove the CLI and tsConfig part of this MR, and just leave the annotation.

yeah, fair enough, I'll do that.